torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -381,6 +381,48 @@ std::future<bool> launchAsyncGilCheck() {
   return resultFuture;
 }
 
+class CUDAEventCache {
+ public:
+  CUDAEventCache() : caches_(at::cuda::device_count()) {}
```

Comment:
ProcessGroupNCCL now supports single device per thread only. Would that help make the implementation here even simpler?